heron/instance/src/java/com/twitter/heron/instance/OutgoingTupleCollection.java

```diff
@@ -41,8 +41,13 @@
   // Total data emitted in bytes for the entire life
   private long totalDataEmittedInBytes = 0;
 
-  private int dataTupleSetCapacity;
-  private int controlTupleSetCapacity;
+  // Current size in bytes for data types to pack into the HeronTupleSet
+  private long currentDataTupleSizeInBytes = 0;
+  // Maximum data tuple size in bytes we can put in one HeronTupleSet
+  private long maxDataTupleSizeInBytes = Long.MAX_VALUE;
```

Comment:
Yes they are always get overwritten in the constructor. Will uniform the initialization in construct in the new commit.